app/src/main/java/org/mozilla/focus/widget/DefaultBrowserPreference.java

```diff
@@ -75,9 +119,120 @@ private void openDefaultAppsSettings(Context context) {
+        getContext().startActivity(viewIntent);
+    }
+
+    /**
+     * To define necessary actions for setting default-browser.
+     */
+    private interface DefaultBrowserAction {
+        void onPrefClicked();
+
+        void onFragmentResume();
+
+        void onFragmentPause();
+    }
+
+    private static class DefaultAction implements DefaultBrowserAction {
+        DefaultBrowserPreference pref;
+
+        DefaultAction(@NonNull final DefaultBrowserPreference pref) {
+            this.pref = pref;
+        }
+
+        public void onPrefClicked() {
+            // fire an intent and start related activity immediately
+            pref.openDefaultAppsSettings(pref.getContext());
+        }
+
+        public void onFragmentResume() {
+
+        }
+
+        public void onFragmentPause() {
+        }
+    }
+
+    /**
+     * For android sdk version older than N
+     */
+    private static class LowSdkAction implements DefaultBrowserAction {
+        DefaultBrowserPreference pref;
+        BroadcastReceiver receiver;
+
+        LowSdkAction(@NonNull final DefaultBrowserPreference pref) {
+            this.pref = pref;
+            this.receiver = new ServiceReceiver(pref);
+        }
+
+        public void onPrefClicked() {
+            final Context context = pref.getContext();
+            final boolean isDefaultBrowser = Browsers.isDefaultBrowser(context);
+            final boolean hasDefaultBrowser = Browsers.hasDefaultBrowser(context);
+
+            if (isDefaultBrowser) {
+                pref.openAppDetailSettings(context);
+            } else if (hasDefaultBrowser) {
+                pref.setEnabled(false);
+                pref.setSummary(R.string.preference_default_browser_is_setting);
+                pref.clearDefaultBrowser(context);
+            } else {
+                pref.triggerWebOpen();
+            }
+        }
+
+        public void onFragmentResume() {
+            LocalBroadcastManager.getInstance(pref.getContext())
+                    .registerReceiver(this.receiver, ComponentToggleService.SERVICE_STOP_INTENT_FILTER);
+        }
+
+        public void onFragmentPause() {
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
+                LocalBroadcastManager.getInstance(pref.getContext())
+                        .unregisterReceiver(this.receiver);
+            }
+        }
+    }
+
+    private static class ServiceReceiver extends BroadcastReceiver {
+        DefaultBrowserPreference pref;
+
+        ServiceReceiver(DefaultBrowserPreference pref) {
+            this.pref = pref;
+        }
+
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            // update UI
+            pref.update();
+
+            // SettingsActivity is in foreground(because this BroadcastReceiver is working),
+            // to remove notification which created by Service
+            NotificationManagerCompat.from(context).cancel(ComponentToggleService.NOTIFICATION_ID);
```

Comment:
I'm wondering if the user pref off "Send Usage Data" and then Firebase was also stopped, what will happen here...

Reply:
Nothing is supposed to change here